### PR TITLE
Test/fix tests

### PR DESCRIPTION
**Pull request by talekhinezh**
Fix kernel_fuzzer test by only committing to database if end of transaction transient substates don't contain owned nodes

### CODE REVIEW

File: radix-engine/src/transaction/transaction_executor.rs [hunk 1]
```diff
@@ -382,7 +382,9 @@ where
                             execution_trace_module.finalize(&paying_vaults, is_success);
 
                         // Finalize track
-                        let (tracked_nodes, deleted_partitions) = track.finalize();
+                        let (tracked_nodes, deleted_partitions) = track.finalize().expect(
+                            "System invariants should prevent transient substate from owning nodes",
+                        );
```
Comment (iamyulong) [inline radix-engine/src/transaction/transaction_executor.rs:387]: Can we match the error code and then panic?

File: radix-engine/src/track/track.rs [hunk 2]
```diff
@@ -188,12 +196,19 @@ impl<'s, S: SubstateDatabase, M: DatabaseKeyMapper + 'static> Track<'s, S, M> {
                     .and_then(|tracked_node| tracked_node.tracked_partitions.get_mut(&partition))
                 {
                     let db_sort_key = M::to_db_sort_key(&substate_key);
-                    tracked_partition.substates.remove(&db_sort_key);
+                    let tracked_substate = tracked_partition.substates.remove(&db_sort_key);
+                    if let Some(substate) =
+                        tracked_substate.and_then(|s| s.substate_value.into_value())
```
Comment (iamyulong) [inline radix-engine/src/track/track.rs:201]: Wonder under what scenario the substate is not tracked but marked as transient.

File: transaction-scenarios/generated-examples/non_fungible_resource/005--non-fungible-resource-mint-32-nfts.rtm [hunk 3]
```diff
@@ -4,7 +4,7 @@ CALL_METHOD
     Decimal("5000")
 ;
 MINT_NON_FUNGIBLE
-    Address("resource_sim1ntjqlurzucu3lxnuxu4e7v57qmfna3gxuka5ll2we0k2zz5ptnlmy0")
+    Address("resource_sim1ngx46l2l7t93rpwchzkv2ywdxpjdxvk8lplrqfte4d8pg92szl8vuz")
```
Comment (iamyulong) [inline transaction-scenarios/generated-examples/non_fungible_resource/005--non-fungible-resource-mint-32-nfts.rtm:7]: Interesting that the scenario outputs are changed.

Comment (talekhinezh) [reply transaction-scenarios/generated-examples/non_fungible_resource/005--non-fungible-resource-mint-32-nfts.rtm:7]: I think it just wasn't updated on last commit

File: radix-engine/src/track/track.rs (hunk 2 above)
Comment (talekhinezh) [reply radix-engine/src/track/track.rs:201]: There is no scenario unless the system layer is doing something wonky

File: radix-engine/src/transaction/transaction_executor.rs (hunk 1 above)
Comment (talekhinezh) [reply radix-engine/src/transaction/transaction_executor.rs:387]: Yup good point

**Review by iamyulong — APPROVED** [review]

**Pull request merged**